lib/cisco_node_utils/ace.rb

```diff
@@ -192,5 +232,99 @@ def dst_port
     def dst_port=(src_port)
       @set_args[:dst_port] = src_port
     end
+
+    def tcp_flags
+      match = ace_get
+      return nil if match.nil?
+      match.names.include?('tcp_flags') ? match[:tcp_flags] : nil
+    end
+
+    def tcp_flags=(tcp_flags)
+      @set_args[:tcp_flags] = tcp_flags
+    end
+
+    def established
+      match = ace_get
+      return nil if match.nil?
+      match.names.include?('established') ? true : false
+    end
+
+    def established=(established)
+      @set_args[:established] = established == true ? 'established' : ''
+    end
+
+    def precedence
+      extract_value('precedence', 'precedence')
+    end
+
+    def precedence=(precedence)
+      attach_prefix(precedence, :precedence, 'precedence')
+    end
+
+    def dscp
+      extract_value('dscp', 'dscp')
+    end
+
+    def dscp=(dscp)
+      attach_prefix(dscp, :dscp, 'dscp')
+    end
+
+    def time_range
+      extract_value('time_range', 'time-range')
+    end
+
+    def time_range=(time_range)
+      attach_prefix(time_range, :time_range, 'time-range')
+    end
+
+    def packet_length
+      extract_value('packet_length', 'packet-length')
+    end
+
+    def packet_length=(packet_length)
+      attach_prefix(packet_length, :packet_length, 'packet-length')
+    end
+
+    def ttl
+      extract_value('ttl', 'ttl')
+    end
+
+    def ttl=(ttl)
+      attach_prefix(ttl, :ttl, 'ttl')
+    end
+
+    def http_method
+      extract_value('http_method', 'http-method')
+    end
+
+    def http_method=(http_method)
+      attach_prefix(http_method, :http_method, 'http-method')
+    end
+
+    def tcp_option_length
+      extract_value('tcp_option_length', 'tcp-option-length')
+    end
+
+    def tcp_option_length=(tcp_option_length)
+      attach_prefix(tcp_option_length, :tcp_option_length, 'tcp-option-length')
+    end
+
+    def redirect
+      extract_value('redirect', 'redirect')
+    end
+
+    def redirect=(redirect)
+      attach_prefix(redirect, :redirect, 'redirect')
+    end
+
+    def log
+      match = ace_get
+      return nil if match.nil?
+      match.names.include?('log') ? true : false
+    end
+
+    def log=(log)
+      @set_args[:log] = log == true ? 'log' : ''
```

Comment:
The correction will set log when empty string is passed as argument, which we do not want.